### PR TITLE
add: imagePullSecret support for private container registry

### PR DESCRIPTION
**Pull request by tylern91**
Signed-off-by: Tyler Nguyen <tyler.nguyen1191@gmail.com>

This PR adds support for `imagePullSecrets` argument, which is required for the use of mirrored image from the private container registry.

### CODE REVIEW

**Comment by tylern91** [comment]
@mureg This replaces the previous closed PR which the sync from the chart. PTAL

**Review by zouyee — APPROVED** [review]

**Comment by mrueg** [comment]
@tylern91 not having a strong preference here, other charts would add the pullsecrets to the serviceaccount instead of the deployment. Any thoughts on why you added them to the deployment objects?

**Comment by tylern91** [comment]
> @tylern91 not having a strong preference here, other charts would add the pullsecrets to the serviceaccount instead of the deployment. Any thoughts on why you added them to the deployment objects?

Well, it depends on how we would like to manage those attributes in terms of the balance between fine-drained management with more secured and flexibility.

Updated to serviceaccount by https://github.com/coredns/helm/pull/34/commits/2c9e44a3ae8b46e6e1ac18be9987c91a94efe4f7

**Comment by mrueg** [comment]
looks good to me. Can you rebase on latest master?

**Comment by tylern91** [comment]
> looks good to me. Can you rebase on latest master?

Done by  https://github.com/coredns/helm/pull/34/commits/2f77a4229f7d4697e6827ea85378e4dbf2bb9ca1 ✅

**Comment by mrueg** [comment]
It still conflicts with master and can't be merged

**Comment by tylern91** [comment]
@mrueg I didn't see any conflict with the master. Where does it show the conflict? 🤔

**Comment by mrueg** [comment]
```This branch cannot be rebased due to conflicts
Rebasing the commits of this branch on top of the base branch cannot be performed automatically due to conflicts encountered while reapplying the individual commits from the head branch. ``` is what I see here.

File: stable/coredns/values.yaml [hunk 1]
```diff
@@ -275,6 +287,15 @@ autoscaler:
     # i.e. strategy.spinnaker.io/versioned: "false" to ensure configmap isn't renamed
     annotations: {}
 
+  # Enables the livenessProbe for cluster-proportional-autoscaler - this requires version 1.8.0+ of the autoscaler
```
Comment (mrueg) [inline stable/coredns/values.yaml]: this probably should not be in here as it was merged in a separate PR

Comment (tylern91) [reply stable/coredns/values.yaml:290]: Closed and replaced by https://github.com/coredns/helm/pull/36

**Pull request closed without merging**